impl/random.h

```diff
@@ -99,6 +97,58 @@ hydro_random_init(void)
     return 0;
 }
 
+#elif (defined(NRF52832_XXAA) || defined(NRF52832_XXAB)) && !defined(__unix__)
+
+// Important: The SoftDevice *must* be activated to enable reading from the RNG
+// http://infocenter.nordicsemi.com/index.jsp?topic=%2Fcom.nordic.infocenter.nrf52832.ps.v1.1%2Frng.html
+
+#include <nrf_soc.h>
+
+static int
+hydro_random_init(void)
+{
+    const char       ctx[hydro_hash_CONTEXTBYTES] = { 'h', 'y', 'd', 'r', 'o', 'P', 'R', 'G' };
+    hydro_hash_state st;
+
+    const uint8_t    total_bytes = 32;
+    uint8_t          remaining_bytes = total_bytes;
+    uint8_t          available_bytes;
+
+    uint8_t          rand_buffer[total_bytes];
+
+    hydro_hash_init(&st, ctx, NULL);
+
+    while (true) {
+        if (sd_rand_application_bytes_available_get(&available_bytes) != NRF_SUCCESS) {
+            return -1;
+        }
+
+        if (available_bytes > 0) {
+            if (available_bytes > remaining_bytes) {
+                available_bytes = remaining_bytes;
+            }
+
+            if (sd_rand_application_vector_get(rand_buffer, available_bytes) != NRF_SUCCESS) {
+                return -1;
+            }
+
+            hydro_hash_update(&st, rand_buffer, total_bytes);
+            remaining_bytes -= available_bytes;
+        }
+
+        if (remaining_bytes > 0) {
```

Comment:
Are you should it shouldn't be `<= 0` here?

Reply:
Yes, because `remaining_bytes` counts down as bytes become available from the random pool. If not enough bytes were available on this iteration, then there are still bytes "remaining" before we've collected a full 32, so we delay and loop.